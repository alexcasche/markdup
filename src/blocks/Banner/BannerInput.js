import styled from 'styled-components'
import { tint } from 'polished'

const BannerInput = styled.input`
  width: 100%;
  margin-top: 6px;
  padding: 0;
  background-color: transparent;
  border: none;
  outline: none;
  color: ${props => tint(0.95, props.theme.colors.grey)};
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 20px;
`

export default BannerInput
