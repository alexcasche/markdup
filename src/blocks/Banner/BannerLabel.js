import styled from 'styled-components'
import { shade } from 'polished'

const BannerLabel = styled.label`
  display: block;
  color: ${props => shade(0.85, props.theme.colors.grey)};
  font-size: 11px;
  text-transform: uppercase;
  letter-spacing: 2px;
`

export default BannerLabel
