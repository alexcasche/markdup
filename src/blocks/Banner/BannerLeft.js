import styled from 'styled-components'

const BannerLeft = styled.div`
  flex-basis: 50%;
  padding-right: 10px;
`

export default BannerLeft
