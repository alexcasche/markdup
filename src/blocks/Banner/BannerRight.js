import styled from 'styled-components'

const BannerRight = styled.div`
  flex-basis: 50%;
  padding-left: 10px;
  text-align: right;
`

export default BannerRight
