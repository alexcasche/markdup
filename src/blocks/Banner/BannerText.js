import styled from 'styled-components'
import { shade, tint } from 'polished'

const BannerText = styled.p`
  color: ${props => shade(0.85, props.theme.colors.grey)};
  font-size: 13px;
  font-weight: 500;
  line-height: 20px;
  letter-spacing: 2px;
  text-transform: capitalize;
  span {
    margin-left: 5px;
    color: ${props => tint(0.95, props.theme.colors.grey)};
    font-weight: 400;
    letter-spacing: 1px;
  }
`

export default BannerText
