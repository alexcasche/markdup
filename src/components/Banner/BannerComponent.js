import React from 'react'
import { Banner } from 'blocks'
import { PropTypes as t } from 'prop-types'
import getCount from './getCount'

const BannerComponent = props => {
  const { activeEditor, currentWorkspace, actions } = props
  const count = getCount(activeEditor, currentWorkspace)
  return (
    <Banner className="Banner">
      <Banner.Left>
        <Banner.Label for="Banner-Input">workspaces</Banner.Label>
        <Banner.Input
          id="Banner-Input"
          value={currentWorkspace.title}
          onChange={e => actions.updateWorkspace('title', e.target.value)}
        />
      </Banner.Left>
      <Banner.Right>
        <Banner.Text>
          lines: <span>{count.lines}</span>
          <br />
          characters: <span>{count.characters}</span>
        </Banner.Text>
      </Banner.Right>
    </Banner>
  )
}

BannerComponent.propTypes = {
  activeEditor: t.string.isRequired,
  currentWorkspace: t.shape({
    title: t.string,
    markdown: t.string,
    html: t.string,
  }).isRequired,
  actions: t.shape({
    updateWorkspace: t.func.isRequired,
  }).isRequired,
}

export default BannerComponent
