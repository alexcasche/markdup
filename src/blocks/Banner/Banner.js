import styled from 'styled-components'
import { tint } from 'polished'

import BannerInput from './BannerInput'
import BannerLabel from './BannerLabel'
import BannerLeft from './BannerLeft'
import BannerRight from './BannerRight'
import BannerText from './BannerText'

const Banner = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 80px;
  padding: 0 20px;
  background-color: ${props => tint(0.95, props.theme.colors.black)};
  font-family: ${props => props.theme.text.fontSans};
`
Banner.Input = BannerInput
Banner.Label = BannerLabel
Banner.Left = BannerLeft
Banner.Right = BannerRight
Banner.Text = BannerText

export default Banner
