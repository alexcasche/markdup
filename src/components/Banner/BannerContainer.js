import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getSettingVal, getCurrentWorkspace } from 'selectors'
import { updateWorkspace } from 'actions'
import BannerComponent from './BannerComponent'

export const BannerContainer = props => {
  return <BannerComponent {...props} />
}

function mapStateToProps(state) {
  return {
    activeEditor: getSettingVal(state, 'activeEditor'),
    currentWorkspace: getCurrentWorkspace(state),
  }
}
function mapDispatchToProps(dispatch) {
  return {
    actions: {
      updateWorkspace: bindActionCreators(updateWorkspace, dispatch),
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BannerContainer)
