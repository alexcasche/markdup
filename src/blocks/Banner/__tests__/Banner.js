import Banner from '../Banner'

describe('Banner Block', () => {
  it('renders correctly', () => {
    const tree = render(<Banner />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
