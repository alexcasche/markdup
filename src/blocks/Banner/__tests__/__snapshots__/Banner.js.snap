// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Block renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 80px;
  padding: 0 20px;
  background-color: #282a2c;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
}

<div
  className="c0"
/>
`;
